### PR TITLE
feat: Backlog/unreal/project context/shot publisher

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves : 

* CLICKUP-865bkucw2

This PR has been tested on :

- [X] Windows.
- [ ] MacOs.
- [ ] Linux.

## Changes

- Give artist ability to select a master sequence with shots, and publish them with batch publisher as image sequences and reviewables.

## Test

_Note: Must be tested with corresponding branch on definition, qt and unreal._

1. Create a master sequence with proper animated shots in Unreal.
2. Render the shots to a folder, with {shot} suffix to path.
3. Select the sequence in Level and run the ftrack Batch publisher.
4. Select a sequence from ftrack, and the folder rendered to (capture folder).
5. Run the batch publisher, shots should be created in ftrack with image sequence/movie asset versions.

### CODE REVIEW

File: resource/definitions/publisher/unreal/shot-unreal-publish.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
+  "type": "publisher",
+  "name": "Shot Publisher",
+  "asset_type": "img",
+  "host_type": "unreal",
+  "ui_type": "qt",
+  "batch": true,
```
Comment (lluisCM) [inline resource/definitions/publisher/unreal/shot-unreal-publish.json:7]: What means batch = True?

File: resource/definitions/schema/publisher-schema.json [hunk 2]
```diff
@@ -568,10 +568,11 @@
     "components",
     "finalizers",
     "type",
+    "batch",
     "_config"
   ],
   "order":[
-    "type", "name", "asset_type", "task_type", "host_type", "ui_type", "discoverable" ,"contexts", "components", "finalizers"
+    "type", "name", "asset_type", "task_type", "host_type", "ui_type", "batch", "discoverable" ,"contexts", "components", "finalizers"
```
Comment (lluisCM) [inline resource/definitions/schema/publisher-schema.json:575]: Why do we need to identify the definitions with batch?

File: resource/plugins/unreal/python/publisher/context/unreal_sequence_publisher_context.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    plugin_name = 'unreal_sequence_publisher_context'
+
+    def run(self, context_data=None, data=None, options=None):
+        '''Find out the sequence context id and shot name and sync the shot to ftrack'''
+
+        shot_name = options.get('shot_name')
+        sequence_context_id = options.get('sequence_context_id')
+
+        try:
+            shot = unreal_utils.push_shot_to_server(
```
Comment (lluisCM) [inline resource/plugins/unreal/python/publisher/context/unreal_sequence_publisher_context.py:25]: IMO this should be a plugin that publish the selected shot. So the batch publisher will collect an amount of shots and will start running the shot publisher definition for each shot.

**Review by lluisCM — CHANGES_REQUESTED** [review]
See the comments.

**Review by lluisCM — APPROVED** [review]
Approving this for the alpha version, but I think this code should never be released, we should clean it out and try to make it more maintainable.

File: resource/definitions/publisher/unreal/shot-unreal-publish.json (hunk 1 above)
Comment (henriknorin-ftrack) [reply resource/definitions/publisher/unreal/shot-unreal-publish.json:7]: This denotes that publisher is only for batch publisher and is not defined for standard single version publisher.

File: resource/definitions/schema/publisher-schema.json (hunk 2 above)
Comment (henriknorin-ftrack) [reply resource/definitions/schema/publisher-schema.json:575]: See comment above

File: resource/plugins/unreal/python/publisher/context/unreal_sequence_publisher_context.py (hunk 3 above)
Comment (henriknorin-ftrack) [reply resource/plugins/unreal/python/publisher/context/unreal_sequence_publisher_context.py:25]: This is the plugin - unreal_sequence_publisher_context.py - publishes sequence (e.g. an Unreal shot)

**Pull request merged**